### ISSUE TITLE
Advice Generating dmabuf GL Texture from Secondary Thread!?

### ISSUE THREAD

**Issue by Fredrum** [libsdl-org/SDL#5259]
Hello!

Iv'e got a program that's taking streaming h264 video data and displaying in a SDL Gl window.
There's a secondary thread that starts up a process that on callback initiates ffmpeg/avcodec decode and then GL texture generation with,
```
eglCreateImageKHR( egl_display,
                                              EGL_NO_CONTEXT,
                                              EGL_LINUX_DMA_BUF_EXT,
                                              NULL,
                                              attribs);
```
The decode + gl render is designed to be kept 'memory-to-memory' ie no copy.

Latency is top priority.

My problem is that I think Iv'e identified that doing the texture gen from the secondary thread does not work with OpenGL and I believe it has to do with Gl contexts? 
Iv'e tried playing with _SDL_GL_CreateContext(sdl_window)_ and  re-setting and swapping those contexts using SDL but so far without any success.

I have verified that otherwise the code to do with the GL render pipe is working.

Does anyone have advice for what general approach I need to take for this to work with minimum latency?
Basically how I need to set up the gl contexts and switching?

Cheers!

**Comment by cgutman** [comment]
I'd just pass the AVFrame to your main thread (either by pushing a `SDL_UserEvent` or your own queue) then call `eglCreateImageKHR()` with your existing GL context.

**Comment by Fredrum** [comment]
Thanks for the suggestion.
I started thinking along those lines a bit later but wasn't sure how I can be sure the frame gets converted/uploaded asap and not waiting for a frame flip or something like that in my inner while(1) draw loop.

I will go with your suggestion and see what I can do.

**Comment by cgutman** [comment]
The frame is _already_ uploaded. That's the magic of using `EGL_EXT_image_dma_buf_import`.

`eglCreateImageKHR()` and `glEGLImageTargetTexture2DOES()` are cheap because there's no upload needed.
